website/static/js/profile.js

```diff
@@ -463,23 +490,54 @@ var extendLink = function(obs, $parent, label, baseUrl) {
     return obs;
 };
 
-var SocialViewModel = function(urls, modes) {
-    var self = this;
-    BaseViewModel.call(self, urls, modes);
-    TrackedMixin.call(self);
+var extendProfileWebsite = function(obs, $parent, label, baseUrl) {
+    
+    obs.url = ko.computed(function($data, event) {
+        // Prevent click from submitting form
+        event && event.preventDefault();
+        if (obs()) {
+            return baseUrl ? baseUrl + obs() : obs();
+        }
+        return '';
+    });
 
-    self.addons = ko.observableArray();
 
-    self.personal = extendLink(
+    return obs;
+};
+
+
+function ProfileWebsite() {
```

Comment:
Why is this constructor necessary? Can `profileWebsites` just be an array of 'extended' observables?